### DISCUSSION TITLE
XSD, standard adókód véleményezés

### DISCUSSION THREAD

**Discussion by bmatyi94** [nav-gov-hu/eVAT#29] [Ötletek - gépi interfész]
Tisztelt Illetékesek!

Egy két standard adókóddal kapcsolatban úgy tűnt számunkra, hogy nem megfelelően lett generálva a standard adókód. Csatoltuk az általunk helyesnek vélt verziót, a kódok szöveges leírása alapján.
[Hibas adokodok_v1.0.xlsx](https://github.com/nav-gov-hu/eVAT/files/11004353/Hibas.adokodok_v1.0.xlsx)

XSD-vel és a kiadott dokumentációval kapcsolatos észrevételeink:

Önellenőrzés (számos dolog leképzésre került a jelenlegi XSD-ben a bevallási nyomtatvány alapján, az alábbi 2 kivétellel. Érthető, hogy az önellenőrzési funkció az eÁFA bevezetésével még nem fog életbe lépni, viszont a sémában érdemes lehet átgondolni a működését):

  - Bevallás jellege (önellenőrzés vagy helyesbítés) jelölését nem találtuk az XSD sémában. Ez a jövőben nem lesz szükséges?
  - Önellenőrzés esetén az eredeti adó összegének megadás nem része az XSD-nek. Ennek oka, hogy amúgy is ott lesz az Adóhatóságnál az adat?

XSD tag kötelezőségek:
 - Amennyiben nincs átvihető ÁFA a következő időszakra, vagy előző időszakról beszámítható csökkentő tétel, javasolt ezen tag-eket nem kötelezővé tenni. Természetesen a 0 érték is beírható, de ha lehetséges, akkor javasoljuk, hogy ne legyen kötelező a sumResidualTax és sumTransfarable értéke. 
 
 - Több társaságnál tapasztaljuk a gyakorlatban, hogy az áfa analitikájuk nem tartalmazza (nem került az adat rögzítésre) levonható oldalon a teljes adóalapot és áfa összeget adókulcs szerinti megbontásban, hanem csak a levonható áfa összegét és a hozzá tartozó adóalapot. Továbbá a levonási hányadost sem minden esetben. Levonható oldalon a bevallás szempontjából a deductionAmount fog számítani és nem a tax amount ami az adott tétel teljes áfa összegére vonatkozna adókulcs szerinti megbontásban. 
 
 Véleményünk szerint ezen követelmény, amely alapján az Adóhatóság megköveteli a teljes adóalap, áfa összeget (ráadásul ezt standard adókód megbontásában, azaz ha a számla több adókulcsot tartalmaz akkor az egyes adókulcsokhoz tartozó teljes adóalap és áfa összegére van szükség, nem számlaszinten van szükség a számla adóalapjára és áfa összegére) és levonható adót, valamint a levonási hányadost az eÁFA M2M rendszer belépési küszöb szintjét megemeli. Mivel az Adóhatóság a standard adókódokat a bevallási sorok alapján és nem ügylettípusok alapján képezte le, így egyértelmű információval nem fog rendelkezni, hogy az adott számlatétel például egy telefonszolgáltatás tételt, vagy egy lízingszolgáltatás áfáját tartalmazza, így a megadott levonási hányados alapján a teljes áfa és a levont áfa közötti matematikai ellenőrzést tud feltételezésem szerint végezni, hogy helyes-e a levont áfa összege a levonási hányados és a teljes áfa szorzata alapján. Ezen matematikai validáció nem képez akkora hozzáadott értéket, mint amekkora elvárást támaszt a részletes adatok megadásával szemben.

Bevallási lapok:
- ha jól értem a sheetFieldName (earData xsd) elem az ÁNYK BEVFELD mező értékéhez kapcsolódik. Ez logikailag zavaros, hogy a fejlesztőknek miért kéne ismerniük az eÁFA M2M rendszerben az ÁNYK mezőazonosítókat, főleg, hogy előbb utóbb az ÁNYK rendszer kivezetésre kerül. Bevallási lapok elemeit is érdemes lenne az XSD struktúrában kibontani.

Teljesítés dátum hiánya:
- Taxpoint date nem minden esetben egyezik meg a teljesítés időpontjával, általánosságban igen, de vannak ügyletek ahol az áfafizetési kötelezettség időpontja nem a teljesítés időpontjára épül. Feltételezem az Adóhatóság céljai között szerepel, a helyes áfa bevallási időszak ellenőrzése is. 

Egyes társaságoknál a vállalatirányítási rendszerük tartalmazza a taxpoint date-t, de ebben lehet hiba is. A dokumentum kiállításának dátuma mellett a teljesítési dátum megadásával a taxpoint date adata már jobban kontrollálható, ellenőrizhető lenne. Legalábbis segítheti az adózókat például, ha olyan számla nem kerülne be a bevallásba, aminek az áfáját esetleg korábbi időszakra kellett volna megfizetni és bevallani és erről akár kaphatnának validációs üzenetet.

Teljes áfa bevallás leképzése:
Végezetül egy személyes vélemény. Érthető az Adóhatóság logikája, hogy a dataXML lényegében nagyjából áfa bevallás leképzése, az áfa analitikával kiegészítve. Azonban ez azt is jelenti, hogy minden szoftveres megoldásnak le kell képeznie egy olyan grafikus felületet, melyen a bevallási információk (például az áfa visszaigénylésről való döntés)is megadhatóak, hogy a dataXML részét tudják képezni. Minden elvárt elem nem automatizálható, hiszen vannak pontok, ahol a felhasználó, adózó döntésére van szükség. Vannak rendszerek, ahol ez már most is le van képezve és nem jelent gondot beépíteni a dataXML-be. Viszont nem biztos, hogy célszerű, hogy akiknél ez jelenleg nem áll rendelkezésre ők is kötelezően képezzék le a XX65 bevallás egy részét, amikor ez központilag is támogatható lenne az Adóhatóság oldaláról. Érdemes lehet megfontolni, azt a megközelítést is, hogy az áfa analitika kerül beküldésre M2M kapcsolattal a Hatósághoz, viszont a bevallási információk megadása az Adóhatóság eÁFA felületén lenne megtehető. Így azon cégeknél, ahol megtudják oldani vagy már megoldott a grafikus felület az egyes adózói döntésekhez a bevallási adatokkal kapcsolatban, és a felhasználók nem akarnak kilépni az általuk használt szoftverből, ott a dataXML elkészíthető és beküldhető a bevallási adatokkal együtt. Viszont esetleg lehetőséget kaphatnának azon társaságok is, akiknek az áfa analitika rendelkezésre állna az elvárt XML formátumban, azonban egy új grafikus felület lefejlesztése helyett, az Adóhatóság eÁFA honlapján fejezhetnék be a bevallásukat.

Köszönöm a javaslatok, észrevételek átnézését.

Üdvözlettel:

Budai Mátyás
RSM Hungary Zrt.

**Comment by bmatyi94** [comment]
Plusz még egy javaslat:

Ügyfélkörben több esetben tapasztaltuk, hogy az áfa analitikák tartalmaztak könyvelési korrekciókat. Például amennyiben rossz adókód lett párosítva az adott számlához, akkor a számla adata megjelenik a rossz adókódhoz párosítva, majd egy mínuszos sor ugyan azon adatokkal (mely kiüti a rossz adókód besorolást), ezután megjelenik a jó adókód párosítással a számlaadat. Ezen könyvelési korrekciók mögött ténylegesen nem állnak számlák. Érdemes lehet a SourceDocumentType-nál erre is bevezetni enumerationt. (Other is tulajdonképpen egyéb bizonylatot takar a jelenlegi sémában, így ez a könyvelési korrekció nincs levedve, holott nagyon gyakori, bár igaz az is lehet a megközelítés hogy ezen típusú tételeket ne tegyék be a társaságok a dataXML-be). Amennyiben ez mégse kerülne lekezelésre, érdemes validációt párosítani hozzá, hiszen például az online számla összehasonlítás az említett mínuszos résznél nem értelmezhető.

**Comment by bmatyi94** [comment]
Tisztelt Adóhatóság! @NTCA-developer @NTCA-HM @NTCA-tax 

Az ÁFA-rendszer gép-gép kapcsolathoz készített fejlesztői dokumentációt, valamint a hozzájuk tartozó XSD-sémák áttekintése után is a fentebbi véleményünket fenntartjuk.

**Comment by renced42** [comment]
Kedves @bmatyi94 

Köszönjük az észrevételeket.

Az önellenőrzés fejlesztése folyamatban van azt az eÁfa rendszerben később lehet majd megtenni.
A könyvelésben elrontott tételeket a NAV nem szeretné látni, így be sem szeretnénk kérni. 
A többi felvetéssel kapcsolatosan az egységességre törekszünk, kivételeket nem igazán szeretnénk kezelni. Könyvelés technikailag, biztosan vannak eltérő módszerek reméljük, hogy ezekkel ezek is egységesednek.
